### PR TITLE
Add Calculation of the entire Capacitance Matrix

### PR DESCRIPTION
**Pull request by lmh91**
In this PR the calculation of the capacitance matrix is added.

The individual elements, c_{i,j}, are calculated via their weighting potentials, WP_i and WP_j, 
as derived [elsewhere](https://www.jpier.org/PIERB/pierb92/01.21011501.pdf).
I also added the formula to the docstring.

I adjusted the previous tests and added another one for the analytically solvable case of two spherical conductors
also described [elsewhere](https://www.comsol.com/blogs/how-to-calculate-a-capacitance-matrix-in-comsol-multiphysics/).

### CODE REVIEW

File: docs/src/man/capacitances.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+c_{ij} = \epsilon_0 \int_{World} \nabla \Phi_i^w(\vec{r}) ϵ_r(\vec{r}) \nabla \Phi_j^w(\vec{r}) d\vec{r}
+```
+
+The function returns the elements of the Maxwell capacitance matrix, $\mathbf{C}$,
+which fulfills
+```math
+\vec{\mathbf{Q}} = \mathbf{Q} \cdot \vec{\mathbf{V}}
+```
+
+The hole matrix can also be obtained via
```
Comment (fhagemann) [inline docs/src/man/capacitances.md:20]: Typo: whole

File: examples/example_config_files/two_spheres_capactior.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: Charged Sphere
```
Comment (fhagemann) [inline examples/example_config_files/two_spheres_capactior.yaml:1]: The name of this file has a type: capacitor

File: src/Simulation/Capacitance.jl [hunk 3]
```diff
@@ -1,146 +1,125 @@
-function _get_abs_bias_voltage(det::SolidStateDetector{T}) where {T <: SSDFloat}
-    potentials::Vector{T} = map(c -> c.potential, det.contacts)
-    return (maximum(potentials) - minimum(potentials)) * u"V"
-end
-
+export calculate_mutual_capacitance
```
Comment (fhagemann) [inline src/Simulation/Capacitance.jl:1]: Should this go into `SolidStateDetectors.jl` where we gather all other `export` statements?

File: src/Simulation/Capacitance.jl [hunk 4]
```diff
@@ -1,146 +1,125 @@
...
-
-    In general, use [`calculate_capacitance(::Simulation, ::Int)`](@ref) 
-    to calculate the capacitance of a specific contact via its weighting potential.
-"""
-function calculate_capacitance(sim::Simulation; consider_multiplicity::Bool = true)
-    W = calculate_stored_energy(sim; consider_multiplicity)
-    return uconvert(u"pF", 2 * W / (_get_abs_bias_voltage(sim.detector)^2))
-end
+!!! note
+    These are elements of the Mawell Capcitance Matrix. Lock up [Capacitances](@ref) for more information.
```
Comment (fhagemann) [inline src/Simulation/Capacitance.jl:13]: Typo: Look

File: src/Simulation/ElectricFieldEnergy.jl [hunk 5]
```diff
@@ -0,0 +1,69 @@
+export calculate_stored_energy
```
Comment (fhagemann) [inline src/Simulation/ElectricFieldEnergy.jl:1]: Same here: Move this to `SolidStateDetectors.jl`?

File: src/examples.jl [hunk 6]
```diff
@@ -24,6 +24,7 @@ sim = Simulation(path_to_config_file)
 ```
 """
 const SSD_examples = Dict{Symbol,String}([
+    :TwoSpheresCapactior                     => joinpath(get_path_to_example_config_files(), "two_spheres_capactior.yaml"),
```
Comment (fhagemann) [inline src/examples.jl:27]: Typo in the file name

File: test/comparison_to_analytic_solutions.jl [hunk 7]
```diff
@@ -18,24 +18,66 @@ e = SolidStateDetectors.elementary_charge * u"C"
...
+        u = acosh((d^2 - r1^2 - r2^2) / (2*r1*r2))
+        for i in 1:n
+            c += inv(sinh(i*u))
+        end
+        uconvert(u"pF", -F * r1 * r2 * sinh(u) * c / d)
+    end
+    A_c11 = c_ii(R_1, R_2, d)
+    A_c22 = c_ii(R_2, R_1, d)
+    A_c12 = c_ij(R_1, R_2, d)
+    C_Analytical = [A_c11 A_c12; A_c12 A_c22]
```
Comment (fhagemann) [inline test/comparison_to_analytic_solutions.jl:66]: Is `C_Analytical` used anywhere?

**Review by fhagemann — COMMENTED** [review]
I just had a quick look over this PR and it looks very promising!
Here are my (very minor) comments

File: test/comparison_to_analytic_solutions.jl (hunk 7 above)
Comment (lmh91) [reply test/comparison_to_analytic_solutions.jl:65]: Now it is.

**Comment by lmh91** [comment]
Thanks! I fixed the typos.

**Comment by fhagemann** [comment]
Very nice, thanks!

**Pull request merged**